### ISSUE TITLE
large_tuple 0.49.1 not catching

### ISSUE THREAD

**Issue by khoogheem** [realm/SwiftLint#4315]
### New Issue Checklist

- [ x] Updated SwiftLint to the latest version
- [ x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```
let blah: (Bool, Bool, Bool, Bool, Bool)
let _: (Int, Int, Int, Bool, Bool, Bool, Bool, Bool)
var (r1, g1, b1, a1): (CGFloat, CGFloat, CGFloat, CGFloat)
```

none of those get caught by the rule for large tuple

```
    func blah(_ blah: (Bool, Bool, Bool, Bool, Bool)) {
```
Still does get caught

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)? 

  0.49.1 via Homebrew
Same results if I use no config file or use the one that worked prior to 0.49.1

**Comment by marcelofabri** [comment]
That snippet triggers for me in all 3 lines in both  0.49.1 and in `main`. We've rewritten this rule in `main` so it should be more accurate and this should trigger, but wanted to make sure I was testing agains the code that failed to trigger in 0.49.1.

**Comment by khoogheem** [comment]
something is odd then.. as with my 0.49.1 via Homebrew it does not trigger
and if I add in `// swiftlint:disable:this large_tuple` I get a warning that this is not needed as it does not trigger

**Comment by SimplyDanny** [comment]
For the file (`main.swift`)
```swift
let blah: (Bool, Bool, Bool, Bool, Bool)
let _: (Int, Int, Int, Bool, Bool, Bool, Bool, Bool)
var (r1, g1, b1, a1): (CGFloat, CGFloat, CGFloat, CGFloat)

func foo(_ blah: (Bool, Bool, Bool, Bool, Bool)) {}
```
0.49.1 installed via Homebrew reports
```
main.swift:1:1: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:3:1: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:3:1: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:3:1: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:3:1: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:5:10: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
```
while `main` reports
```
main.swift:1:11: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:2:8: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:3:23: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
main.swift:5:18: error: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
```
So we clearly see that `main` finds more violations, does not report duplicates and is more accurate in the violation location. However, we also see 0.49.1 finds most of the violations as well except for the one on line 2.

**Comment by khoogheem** [comment]
ok.. I see what is happening..
Yes if I put those examples into main.swift they do in fact get caught correctly..

so it looks like in the function defined vars 0.49.1 is not catching them but some prior (sorry can't recall which version) did

```
    func toColor(_ color: UIColor, percentage: CGFloat) -> UIColor {
        let percentage = max(min(percentage, 100), 0) / 100
        switch percentage {
        case 0:
            return self
        case 1:
            return color
        default:
            var (r1, g1, b1, a1): (CGFloat, CGFloat, CGFloat, CGFloat) = (0, 0, 0, 0)
            var (r2, g2, b2, a2): (CGFloat, CGFloat, CGFloat, CGFloat) = (0, 0, 0, 0)
            guard self.getRed(&r1, green: &g1, blue: &b1, alpha: &a1) else { return self }
            guard color.getRed(&r2, green: &g2, blue: &b2, alpha: &a2) else { return self }
            
            return UIColor(red: CGFloat(r1 + (r2 - r1) * percentage),
                           green: CGFloat(g1 + (g2 - g1) * percentage),
                           blue: CGFloat(b1 + (b2 - b1) * percentage),
                           alpha: CGFloat(a1 + (a2 - a1) * percentage))
        }
    }
```

**Comment by SimplyDanny** [comment]
You are right. 0.49.1 doesn't report anything in this snippet. `main` works well, however. 

The good news is: The coming release is going to fix the issue for you. 😉

**Comment by jpsim** [comment]
You can try out a prerelease here https://github.com/realm/SwiftLint/releases/tag/0.50.0-rc.1

**Comment by jpsim** [comment]
I’m going to close this issue because it’s fixed on Main

**Comment by khoogheem** [comment]
thanks
